akka-http/src/main/scala/akka/http/scaladsl/marshalling/Marshaller.scala

```diff
@@ -126,14 +126,34 @@ object Marshaller
   /**
    * Helper for creating a synchronous [[Marshaller]] to content with a fixed charset from the given function.
    */
-  def withFixedContentType[A, B](contentType: ContentType)(marshal: A ⇒ B): Marshaller[A, B] =
-    strict { value ⇒ Marshalling.WithFixedContentType(contentType, () ⇒ marshal(value)) }
+  def withFixedContentType[A, B](contentType: ContentType)(marshal: A ⇒ B): Marshaller[A, B] = {
+    val f1 = (value: A) ⇒ Marshalling.WithFixedContentType(contentType, () ⇒ marshal(value))
+    val f2 = (_: ExecutionContext) ⇒ (a: A) ⇒ FastFuture.successful(f1(a) :: Nil)
+    new Marshaller[A, B] {
+      def apply(value: A)(implicit ec: ExecutionContext) =
+        try f2(ec)(value)
```

Comment:
Could you inline those functions? The whole execution is strict so preallocation those lambdas is not beneficial.